File: packages/python-sdk/src/blindfold/regex/detectors/se/orgnr.py
"""Swedish Organisationsnummer (organization number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import se_orgnr_checksum


@register_region("se")
class OrgnrDetector(RegexDetector):
    entity_type = EntityType.SE_ORGNR
    score = 0.85
    context_keywords = [
        "organisationsnummer", "orgnr", "org nr", "organization number", "foretag",
    ]
    context_required = True
    context_window = 50

    # 6 digits + optional hyphen + 4 digits
    pattern = re.compile(
        r"\b\d{6}-?\d{4}\b"
    )
    validator = staticmethod(se_orgnr_checksum)

File: packages/js-sdk/src/regex/detectors/se/orgnr.ts
// Swedish Organisationsnummer (organization number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { seOrgnrChecksum } from '../../validators'

class SeOrgnrDetector extends RegexDetector {
  entityType = EntityType.SE_ORGNR
  score = 0.85
  contextKeywords = ['organisationsnummer', 'orgnr', 'org nr', 'organization number', 'foretag']
  contextRequired = true

  pattern = /\b\d{6}-?\d{4}\b/g
  validator = seOrgnrChecksum
}

registerRegion('se', SeOrgnrDetector)

export { SeOrgnrDetector }
